mlrun/api/main.py

```diff
@@ -51,6 +55,78 @@ async def http_status_error_handler(
     )
 
 
+def get_client_addr(scope):
+    # uvicorn expects this to be a tuple while starlette test client sets it to be a list
+    if isinstance(scope.get("client"), list):
+        scope["client"] = tuple(scope.get("client"))
+    return uvicorn.protocols.utils.get_client_addr(scope)
+
+
+@app.middleware("http")
+async def log_request_response(request: fastapi.Request, call_next):
+    request_id = str(uuid.uuid4())
+    silent_logging_paths = [
+        "healthz",
+    ]
+    path_with_query_string = uvicorn.protocols.utils.get_path_with_query_string(
+        request.scope
+    )
+    if not any(
+        silent_logging_path in path_with_query_string
+        for silent_logging_path in silent_logging_paths
+    ):
+        logger.info(
+            "Received request",
+            method=request.method,
+            client_address=get_client_addr(request.scope),
+            http_version=request.scope["http_version"],
+            request_id=request_id,
+            uri=path_with_query_string,
+        )
+    try:
+        response = await call_next(request)
+
+    # Using StarletteHTTPException because
+    # https://fastapi.tiangolo.com/tutorial/handling-errors/#fastapis-httpexception-vs-starlettes-httpexception
+    except StarletteHTTPException as exc:
+        logger.warning(
+            "Request handling returned error status. Sending response",
+            status_code=exc.status_code,
+            request_id=request_id,
+            uri=path_with_query_string,
+            method=request.method,
+        )
+        raise
+    except Exception:
+        logger.warning(
+            "Request handling failed. Sending response",
+            # TODO: When finding a better solution for this turn off access logs (access_log=False in uvicorn.run call)
+            # Basically any exception that not inherits from StarletteHTTPException will return a 500 status code
+            # UNLESS a special exception handler was added for that exception, in this case this exception handler runs
+            # after this middleware and if it will change the response code this 500 won't be true.
+            # currently I can't find a way to execute code after all custom exception handlers so logging unknown status
+            # code and configuring the access log to only print in case of an error
+            status_code="unknown",
```

Comment:
can remove `status_code` from the log kwargs if it's unknown / not determined at this point

Reply:
Fixed to 500